src/extension/VSProject.cs

```diff
@@ -241,16 +241,41 @@ private void Load()
         /// </summary>
         /// <returns>True if the project was successfully loaded, false otherwise.</returns>
         private bool TryLoadDotNetCoreProject()
-        {
+        {
             XmlNode root = _doc.SelectSingleNode("Project");
 
             if (root != null && SafeAttributeValue(root, "Sdk") != null)
             {
                 string targetFramework = _doc.SelectSingleNode("Project/PropertyGroup/TargetFramework").InnerText;
 
                 XmlNode assemblyNameNode = _doc.SelectSingleNode("Project/PropertyGroup/AssemblyName");
-                // Even console apps are dll's even if <OutputType> has value 'EXE'
-                string assemblyName = assemblyNameNode == null ? $"{Name}.dll" : $"{assemblyNameNode.InnerText}.dll";
+                
+                // Even console apps are dll's even if <OutputType> has value 'EXE',
+                // if TargetFramework is netcore
+                string outputType = "dll";
+
+                if (!targetFramework.StartsWith("netcore"))
+                {
+                    // When targetting standard .Net framework, the default is still dll,
+                    // however, OutputType is now honoured.
+                    // Also if Sdk = 'Microsoft.NET.Sdk.Web' then OutputType default is exe
+                    string sdk = root.Attributes["Sdk"].Value;

```

Comment:
The docs below suggest the sdk attribute might not always be present - if it was missing, I think this line would throw a NullReferenceException, wouldn't it? Can we guard against that?

https://docs.microsoft.com/en-us/dotnet/core/tools/csproj#sdk-attribute

Reply:
This whole block of code is conditional on the sdk attribute existing, because otherwise it isn't a .Net Core project - 

```
You need to have the Sdk attribute set to one of those IDs on the <Project> element in order to use the .NET Core tools and build your code.
```